p2p/libp2p/networksharding/kadSharder.go

```diff
@@ -69,11 +69,24 @@ func (ks *kadSharder) GetDistance(a, b sortingID) *big.Int {
 }
 
 // SortList sort the provided peers list
-func (ks *kadSharder) SortList(peers []peer.ID, ref peer.ID) []peer.ID {
-	return sortList(ks, peers, ref)
+func (ks *kadSharder) SortList(peers []peer.ID, ref peer.ID) ([]peer.ID, bool) {
+	sl := getSortingList(ks, peers, ref)
+	// for balance we just try to keep at least 1 connection ouside of shard
+	balanced := len(peers) > inShardCount(sl)
```

Comment:
I think at least one connection would not be enough, why don't we have this as percentage?
float(len(peers)-inShardCount)/float(len(peers))> percentageOutside

where percentageOutside is subunitar e.g 0.2 meaning 20%
same for intrashard

Reply:
Don't we want to also have some connections outside of the shard in the "balanced". Could it be that all peers are in my shard and the kadSharder says it is balanced?

Reply:
Done, we try to keep between 1 connection and 20%

Reply:
Done, we try to keep between 1 connection and 20%